packages/xstate-react/src/createActorContext.tsx

```diff
@@ -1,38 +1,80 @@
 import * as React from 'react';
-import { RestParams, useInterpret } from './useInterpret';
+import { useInterpret } from './useInterpret';
 import { useActor as useActorUnbound } from './useActor';
 import { useSelector as useSelectorUnbound } from './useSelector';
-import { ActorRefFrom, AnyStateMachine, EmittedFrom } from 'xstate';
+import {
+  ActorRefFrom,
+  AnyStateMachine,
+  AreAllImplementationsAssumedToBeProvided,
+  EmittedFrom,
+  InternalMachineOptions,
+  InterpreterOptions,
+  Observer,
+  StateFrom
+} from 'xstate';
+import { UseMachineOptions } from './useMachine';
 
 export function createActorContext<TMachine extends AnyStateMachine>(
   machine: TMachine,
-  ...[options = {}, observerOrListener]: RestParams<TMachine>
+  interpreterOptions?: InterpreterOptions,
+  observerOrListener?:
+    | Observer<StateFrom<TMachine>>
+    | ((value: StateFrom<TMachine>) => void)
 ): {
   useActor: () => ReturnType<typeof useActorUnbound<ActorRefFrom<TMachine>>>;
   useSelector: <T>(
     selector: (snapshot: EmittedFrom<TMachine>) => T,
     compare?: (a: T, b: T) => boolean
   ) => T;
   useActorRef: () => ActorRefFrom<TMachine>;
-  Provider: (props: {
-    children: React.ReactNode;
-    machine?: TMachine | (() => TMachine);
-  }) => React.ReactElement<any, any>;
+  Provider: (
+    props: {
+      children: React.ReactNode;
+      machine?: TMachine | (() => TMachine);
+    } & (AreAllImplementationsAssumedToBeProvided<
+      TMachine['__TResolvedTypesMeta']
+    > extends false
+      ? {
+          options: UseMachineOptions<
+            TMachine['__TContext'],
+            TMachine['__TEvent']
+          > &
+            InternalMachineOptions<
+              TMachine['__TContext'],
+              TMachine['__TEvent'],
+              TMachine['__TResolvedTypesMeta'],
+              true
+            >;
+        }
+      : {
+          options?: UseMachineOptions<
+            TMachine['__TContext'],
+            TMachine['__TEvent']
+          > &
+            InternalMachineOptions<
+              TMachine['__TContext'],
+              TMachine['__TEvent'],
+              TMachine['__TResolvedTypesMeta']
+            >;
+        })
+  ) => React.ReactElement<any, any>;
 } {
   const ReactContext = React.createContext<ActorRefFrom<TMachine> | null>(null);
 
   const OriginalProvider = ReactContext.Provider;
 
   function Provider({
     children,
-    machine: providedMachine = machine
+    machine: providedMachine = machine,
+    options
   }: {
     children: React.ReactNode;
     machine: TMachine | (() => TMachine);
+    options?: any;
```

Comment:
I used `any` for simplicity here since the return type of `createActorContext` is explicitly typed and I didn't want to include the typegen-related monster in both places here. I think that we could drop the explicit return type from there, annotate things in the body and let TS to infer things here. This way each signature would only have to live in one place